File: src_qt/rules/PropertyNode.cpp
#include "PropertyNode.h"
#include "../Brain.h"

namespace CnotiMind
{

	PropertyNode::PropertyNode( const QString& key, const QString& value, Brain* brain, QObject* parent ):
		RuleNode( brain, parent ),
		_key( key ),
		_value( value )
	{

	}

	void PropertyNode::exec()
	{
		_brain->updatePropertyValue( _key, _value );
	}

	void PropertyNode::exec( QHash<QString, QString>& variables )
	{
		// Convert tags that can be in the value of the property
		QString value = _value;
		this->tagsToValue( value, variables );

		_brain->updatePropertyValue( _key, value );
	}

	QString PropertyNode::info( int depth ) const
	{
		return space(depth) + "Property " + _key + ": " + _value;
	}

	PropertyNode *PropertyNode::fromXML(const QString &qName, const QXmlAttributes &atts, Brain *brain, QObject *parent)
	{
		if( qName.compare( "Property", Qt::CaseInsensitive ) == 0 )
		{
			QString value = atts.value( "value" );
			QString name = atts.value( "name" );

			return new PropertyNode( name, value, brain, parent );
		}
		return NULL;
	}

}

File: src_ios/CnotiMind/rules/PropertyNode.m
//
//  PropertyNode.m
//  SingleEmotion
//
//  Created by Gonçalo Rodrigues on 8/31/11.
//  Copyright 2011 Universidade de Coimbra. All rights reserved.
//

#import "PropertyNode.h"
#import "Brain.h"

@implementation PropertyNode

- (id) initWithKeyAndValuAndBrainAndParent:(NSString*)aKey
                                     value:(NSString*)aValue
                                     brain:(Brain*)aBrain 
                                    parent:(id)aParent
{
    if (self = [super initWithBrainAndParent:aBrain parent:aParent]) {
        _key = (aKey == nil) ? [[NSString alloc] initWithString:@""] : [[NSString alloc] initWithString:aKey];
        _value = (aValue == nil) ? [[NSString alloc] initWithString:@""] : [[NSString alloc] initWithString:aValue];    
    }
    return self;
}


- (void) exec
{
    [_brain updatePropertyValue:_key value:_value];
}


- (void) exec:(NSMutableDictionary*)aVariables
{
    // Convert tags that can be in the value of the property
    NSString* value = _value;
    [self tagsToValue:&value variables:&aVariables];
    [_brain updatePropertyValue:_key value:value];
}

- (NSString*) info:(int)aDepth
{
    NSString* info = [NSString stringWithFormat:@""];
    NSString* space = [self space:aDepth];
    
    info = [info stringByAppendingFormat:@"%@ Property key:%@ value:%@", space, _key, _value];
    
    return info;
}

- (void) dealloc
{
    [super dealloc];
}


@end
